### PR TITLE
Reland Clang 11, Roll buildroot to 1bc40a5f916adb93d87e2a2230a0ec43db876c1c

### PR DESCRIPTION
**Pull request by dnfield**
Disables a test on Mac that now fails with the new toolchain due to minor pixel differences.  Test is still enabled on Linux.

Rolls buildroot to flutter/buildroot@1bc40a5f916adb93d87e2a2230a0ec43db876c1c to capture additional warning ignore guards for Xcode builds (e.g. bitcode enablded builds for iOS).

See also: #17451, #17440

### CODE REVIEW

File: shell/platform/embedder/tests/embedder_unittests.cc [hunk 1]
```diff
@@ -2894,6 +2894,9 @@ TEST_F(EmbedderTest, CanQueryDartAOTMode) {
 }
 
 TEST_F(EmbedderTest, VerifyB143464703WithSoftwareBackend) {
+#if !defined(OS_LINUX)
```
Comment (chinmaygarde) [inline shell/platform/embedder/tests/embedder_unittests.cc:2897]: We ensure all tests here run on all platforms. I'd rather have this test disabled (and file a bug with a linked TODO) than skip it this way.

Comment (dnfield) [reply shell/platform/embedder/tests/embedder_unittests.cc:2897]: Golden tests are difficult to get right for all platforms.  We could just skip it outright and re-enable it if we can get Skia gold for engine unit tests, but we'd then lose coverage even on Linux...

Comment (dnfield) [reply shell/platform/embedder/tests/embedder_unittests.cc:2897]: (The actual differences are minor pixel differences in the gradients)

Comment (chinmaygarde) [reply shell/platform/embedder/tests/embedder_unittests.cc:2897]: > Golden tests are difficult to get right for all platforms.

Sigh. Very true. And, looking at this test, I am not sure why the assertions in the next presentation callback are not sufficient. The pixel test here seems to be redundant. 

It's you call on if you want to skip or disable. But please file a bug with a linked TODO either way.

Comment (dnfield) [reply shell/platform/embedder/tests/embedder_unittests.cc]: Added a TODO and filed https://github.com/flutter/flutter/issues/53784

**Comment by dnfield** [comment]
LUCI failures are a known infra issue right now

**Comment by dnfield** [comment]
I'll push the formatting fix once the linux engine builders are fixed.

**Comment by dnfield** [comment]
Going to land this to kick the tree again.  None of the changes impact presubmits compared to the last round of presubmits that passed.

**Pull request merged**